giraffe/src/components/MultiGrid.tsx

```diff
@@ -0,0 +1,689 @@
+  const {width} = props
+
+  const leftGridWidth = state.leftGridWidth
+  const result = width - leftGridWidth
+
+  return result
+}
+
+const cellRendererTopRightGrid = (
+  props: PropsMultiGrid,
+  parent,
+  {columnIndex, ...rest}
+) => {
+  const {cellRenderer, columnCount, fixedColumnCount} = props
+
+  if (columnIndex === columnCount - fixedColumnCount) {
+    return (
+      <div
+        key={rest.key}
+        style={{
+          ...rest.style,
+          width: SCROLLBAR_SIZE_BUFFER,
+        }}
+      />
+    )
+  } else {
+    return cellRenderer({
+      ...rest,
+      columnIndex: columnIndex + fixedColumnCount,
+      parent,
+    })
+  }
+}
+
+const cellRendererBottomLeftGrid = (
+  props: PropsMultiGrid,
+  parent,
+  {rowIndex, ...rest}
+) => {
+  const {cellRenderer, fixedRowCount, rowCount} = props
+
+  if (rowIndex === rowCount - fixedRowCount) {
+    return (
+      <div
+        key={rest.key}
+        style={{
+          ...rest.style,
+          height: SCROLLBAR_SIZE_BUFFER,
+        }}
+      />
+    )
+  } else {
+    return cellRenderer({
+      ...rest,
+      parent,
+      rowIndex: rowIndex + fixedRowCount,
+    })
+  }
+}
+
+const cellRendererBottomRightGrid = (
+  props: PropsMultiGrid,
+  parent,
+  {columnIndex, rowIndex, ...rest}
+) => {
+  const {cellRenderer, fixedColumnCount, fixedRowCount} = props
+
+  return cellRenderer({
+    ...rest,
+    columnIndex: columnIndex + fixedColumnCount,
+    parent,
+    rowIndex: rowIndex + fixedRowCount,
+  })
+}
+
+const columnWidthRightGrid = (state: State, props: PropsMultiGrid, {index}) => {
+  const {columnCount, fixedColumnCount, columnWidth} = props
+  const {scrollbarSize, showHorizontalScrollbar} = state
+
+  // An extra cell is added to the count
+  // This gives the smaller Grid extra room for offset,
+  // In case the main (bottom right) Grid has a scrollbar
+  // If no scrollbar, the extra space is overflow:hidden anyway
+  if (showHorizontalScrollbar && index === columnCount - fixedColumnCount) {
+    return scrollbarSize
+  }
+
+  return typeof columnWidth === 'function'
+    ? columnWidth({index: index + fixedColumnCount})
+    : columnWidth
+}
+
+const rowHeightBottomGrid = (state: State, props: PropsMultiGrid, {index}) => {
+  const {fixedRowCount, rowCount, rowHeight} = props
+  const {scrollbarSize, showVerticalScrollbar} = state
+
+  // An extra cell is added to the count
+  // This gives the smaller Grid extra room for offset,
+  // In case the main (bottom right) Grid has a scrollbar
+  // If no scrollbar, the extra space is overflow:hidden anyway
+  if (showVerticalScrollbar && index === rowCount - fixedRowCount) {
+    return scrollbarSize
+  }
+
+  return typeof rowHeight === 'function'
+    ? rowHeight({index: index + fixedRowCount})
+    : rowHeight
+}
+
+const onScroll = (setState: Function, props: PropsMultiGrid, scrollInfo) => {
+  const {scrollLeft, scrollTop} = scrollInfo
+  setState((prevState: State) => ({
```

Comment:
Because we are using function components, we don't have `this` and so, `onScroll` has to be defined here as a function available to other functions. If any of them need to setState, then it must be passed around.